lib/db/leveldb_dbinstance.go

```diff
@@ -648,6 +677,46 @@ func (db *Instance) AddInvalidToGlobal(folder, device []byte) int {
 	return changed
 }
 
+// removeAbsoluteFiles checks stored files with invalid, absolute paths and removes them.
+func (db *Instance) removeAbsoluteFiles() {
+	t := db.newReadWriteTransaction()
+	defer t.close()
+
+	dbi := t.NewIterator(util.BytesPrefix([]byte{KeyTypeDevice}), nil)
+	defer dbi.Release()
+
+	found := make(map[string]struct{})
+
+	for dbi.Next() {
+		folder := db.deviceKeyFolder(dbi.Key())
+		device := db.deviceKeyDevice(dbi.Key())
+		var f FileInfoTruncated
```

Comment:
I think you can just use `db.deviceKeyName(dbi.Key())` instead of unmarshalling at all here.